packages/iris-grid/src/IrisGrid.tsx

```diff
@@ -3722,11 +3723,13 @@ class IrisGrid extends Component<IrisGridProps, IrisGridState> {
    */
   deleteRanges(ranges: readonly GridRange[]): void {
     const { model } = this.props;
-    this.pending.add(model.delete(ranges)).catch(e => {
-      if (!PromiseUtils.isCanceled(e)) {
-        log.error('Unable to delete ranges', ranges, e);
-      }
-    });
+    if (isDeletableGridModel(model) && model.isDeletable) {
+      this.pending.add(model.delete(ranges)).catch(e => {
+        if (!PromiseUtils.isCanceled(e)) {
+          log.error('Unable to delete ranges', ranges, e);
+        }
+      });
+    }
```

Comment:
This is a case where the `deleteRanges` method should not be called if the model is not deletable. Right now this will just silently fail; instead we want to fail hard (throw an exception) so the caller knows to fix it.
I'd probably write it something like:
```suggestion
    if (!isDeletableGridModel(model) || !model.isDeletable) {
      throw new Error('Model does not support deleting ranges');
    }
    this.pending.add(model.delete(ranges)).catch(e => {
      if (!PromiseUtils.isCanceled(e)) {
        log.error('Unable to delete ranges', ranges, e);
      }
    });
```